### PR TITLE
Export lib/k8s/cluster/getAllowedNamespaces

### LINKED ISSUE #2301: expose getAllowedNamespaces to plugins
There has been a request to use this function from a plugin.

Expose it so it can be used inside a plugin.

Acceptance criteria
- [ ] it should be documented
- [ ] It should be used inside an example plugin somehow

### PR DESCRIPTION
**Pull request by illume**
There was a request to use getAllowedNamespaces in plugins

- **frontend: k8s/cluster: Export getAllowedNamespaces**
- **headlamp-plugin: Bump version to 0.10.0-alpha.2**

Fixes #2301

## How to test

1) Use Headlamp from this branch
2) Install the alpha release of  `npm i @kinvolk/headlamp-plugin@0.10.0-alpha.2` in a plugin
3) Add this into a plugin src/index.tsx and after install the plugin with `npm start`
```ts
import { getAllowedNamespaces } from '@kinvolk/headlamp-plugin/lib/k8s/cluster';
alert(getAllowedNamespaces())
```

4) In Headlamp go to a cluster page, and go to cluster settings and add an 'allowed namespace'
5) Reload cluster page and see alert with the allowed name spaces you just added.

### CODE REVIEW

File: frontend/src/lib/k8s/cluster.ts [hunk 1]
```diff
@@ -17,13 +17,29 @@ import StatefulSet from './statefulSet';
...
+ *          If a zero length list, then no allowed namespace has been configured for cluster.
+ *          If length > 0, allowed namespaces have been configured for this cluster.
+ *
+ * There are usecases where a user doesn't have the authority to list
+ * all the namespaces. In that case it becomes difficult to access things
+ * around Headlamp. To prevent this we can allow the user to pass a set
+ * of namespaces they know they have access to and we can use this set to
+ * make requests to the API server.
+ */
+export function getAllowedNamespaces(cluster?: string | undefined): string[] {
```
Comment (sniok) [inline frontend/src/lib/k8s/cluster.ts:36]: You can assign a default value right in the definition, then you won't need `theCluster` variable
 `getAllowedNamespaces(cluster: string | undefined = getCluster())`

**Review by sniok — APPROVED** [review]
the function can be simplified a bit but it's not major. looks good!

File: frontend/src/lib/k8s/cluster.ts (hunk 1 above)
Comment (illume) [reply frontend/src/lib/k8s/cluster.ts]: Good idea, updated. Thanks.

**Review by skoeva — APPROVED** [review]
LGTM

**Pull request merged**